nameko/containers.py

```diff
@@ -383,7 +403,7 @@ def _run_worker(self, worker_ctx, handle_result):
                 del exc_info
 
                 self.dependencies.all.worker_teardown(worker_ctx)
-                self.dependencies.injections.all.release(worker_ctx)
+                # release?
```

Comment:
It's not safe to use another dependency during teardown (there's no guarantee of order; teardown all happens in parallel) so I don't think there is a use case. The worker is about to be discarded so I don't think it's really required.

Not sure I follow your last sentence though.